scripts/o3de/tests/test_repo_properties.py

```diff
@@ -0,0 +1,180 @@
+#
+# Copyright (c) Contributors to the Open 3D Engine Project.
+# For complete copyright and license terms please see the LICENSE at the root of this distribution.
+#
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+#
+
+import json
+import pytest
+import pathlib
+from unittest.mock import patch
+
+from o3de import repo_properties
+
+TEST_TEMPLATE_REPO_JSON = '''
+{
+    "repo_name": "repotest",
+    "repo_uri": "https://test",
```

Comment:
Let's use a valid URL
```suggestion
    "repo_uri": "https://test.com",
```